code/clue.js

```diff
@@ -1,59 +1,319 @@
+  // Sets time for loader animation and calls pickKiller function after
+  const loaderTime = setTimeout(pickWeapon, 2000);
+  // Hides the first "side"/ p-tag of the card
+  document.getElementById("start-view-card-weapon").style.display = "none";
+  // Hides the killer information
+  document.getElementById('weapon-info-wrapper').style.display = "none";
+  // Displays the loader
+  document.getElementById('loader-wrapper-weapon').style.display = "flex";
+  document.getElementById('weaponCard').style.backgroundColor = '#000'; 
 }
+// Calling the loadingPickWeapon function when clicking the card
+document.getElementById("weaponCard").onclick = loadingPickWeapon;
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//This function will be invoked in the loadingPickRoom function.
+const pickRoom = () => {
+  document.getElementById("loader-wrapper-room").style.display = "none";
+  document.getElementById("room-info-wrapper").style.display = "block";
+  // Randomly selects a room from the array. And add that to the mystery object.
+  mystery.room = randomSelector(rooms)
+  // This will change the styling and the content of the card after it is clicked
+  document.getElementById('roomName').innerHTML = `${mystery.room}`;
+  document.getElementById("start-view-card-room").style.display = "none";
+}
+
+const loadingPickRoom = () => {
+  // Sets time for loader animation and calls pickKiller function after
+  const loaderTime = setTimeout(pickRoom, 2000);
+  // Hides the first "side"/ p-tag of the card
+  document.getElementById("start-view-card-room").style.display = "none";
+  // Hides the killer information
+  document.getElementById('room-info-wrapper').style.display = "none";
+  // Displays the loader
+  document.getElementById('loader-wrapper-room').style.display = "flex";
+  document.getElementById('roomCard').style.backgroundColor = '#000'; 
+}
+// Calling the loadingPickRoom function when clicking the card
+document.getElementById("roomCard").onclick = loadingPickRoom;
+
+
+
+// Function to reveal mystery when clicking the button
+const revealMystery = () => {
+  if (mystery.killer === '') {
+    document.getElementById('mystery').innerHTML = `Please pick a killer!`;
+  } else if (mystery.weapon === '') {
+    document.getElementById('mystery').innerHTML = `Please pick a weapon!`;
+  } else if (mystery.room === '') {
+    document.getElementById('mystery').innerHTML = `Please pick a room!`;
+  } else {
+    document.getElementById('mystery').innerHTML = 
+    `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}!`;
+  }
+}
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// Calling the revealMystery function
+document.getElementById('revealButton').onclick = revealMystery;
```

Comment:
//Your code is well structured and easy to follow.  I like all the comment which helps the reader and yourself understand what the code is doing. Well done with adding extra stuff like shuffling random weapons and loading cards with set time. 👏You could shorten the code a little by combining arrays and objects. like const suspects = [{mrGreen-object},{prPlum.object}...]. Perhaps won't work with the weapons though, since thay are picked randomly from the array.